nexusutils/nexus/nexus.py

```diff
@@ -1104,32 +1126,109 @@ def axis_helper(dim, nxdata, signal, axes, logger):
         )
 
 
+def get_all_is_a_rel_from_hdf_node(hdf_node, logger=None):
+    """Return list of nxdl concept paths for a nxdl element which corresponds to
+    hdf node.
+    """
+    (_, _, elist) = \
+        get_inherited_nodes(None, nx_name=get_nxdl_entry(hdf_node), hdf_node=hdf_node)
+    inheritance_list = []
+    inheritance_list.append(hdf_node_to_self_concept_path(hdf_node, None))
+    if elist:
+        for elem in elist:
+            tmp_path = elem.get('nxdlbase').split('.nxdl')[0]
+            con_path = '/NX' + tmp_path.split('NX')[-1] + elem.get('nxdlpath')
+
+            inheritance_list.append(con_path)
+    if logger is not None:
+        for con in inheritance_list:
+            logger.info(con)
+    else:
+        return inheritance_list
+    return []
+
+
+def hdf_node_to_self_concept_path(hdf_node, logger):
+    """ Get concept or nxdl path from given hdf_node.
+    """
+    # The bellow logger is for deactivatine unnecessary degub message above
+    if logger is None:
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.INFO)
+    (_, _, nxdl_path) = get_nxdl_doc(hdf_node, logger, None)
+    con_path = ''
+    if nxdl_path:
+        for nd_ in nxdl_path:
+            con_path = con_path + '/' + get_node_name(nd_)
+    return con_path
+
+
 class HandleNexus:
     """documentation"""
-    def __init__(self, logger, args):
+    def __init__(self, logger, nexus_file,
+                 d_inq_nd, c_inq_nd):
         self.logger = logger
         local_dir = os.path.abspath(os.path.dirname(__file__))
-        self.input_file_name = args[0] if len(
-            args) >= 1 else os.path.join(local_dir, '../../tests/data/nexus/201805_WSe2_arpes.nxs')
+
+        self.input_file_name = nexus_file if nexus_file is not None else \
+            os.path.join(local_dir, '../../tests/data/nexus/201805_WSe2_arpes.nxs')
         self.parser = None
         self.in_file = None
+        self.d_inq_nd = d_inq_nd
+        self.c_inq_nd = c_inq_nd
+        # Aggregating hdf path corresponds to concept query node
+        self.hdf_path_list_for_c_inq_nd = []
 
     def visit_node(self, hdf_name, hdf_node):
         """Function called by h5py that iterates on each node of hdf5file.
         It allows h5py visititems function to visit nodes."""
-        hdf_path = '/' + hdf_name
-        process_node(hdf_node, hdf_path, self.parser, self.logger)
+        if self.d_inq_nd is None and self.c_inq_nd is None:
+            process_node(hdf_node, '/' + hdf_name, self.parser, self.logger)
```

Comment:
when run from command line, can we set a single logger, so each line is printed only once?